content-resources/src/main/resources/flyway/oskari/V1_46_8__move_externalid_to_separate_table.sql

```diff
@@ -0,0 +1,14 @@
+BEGIN;
+
+SELECT id AS maplayerid, externalid
+INTO oskari_maplayer_externalid
+FROM oskari_maplayer
+WHERE externalid <> '';
+
+ALTER TABLE oskari_maplayer_externalid ADD PRIMARY KEY (maplayerid);
+ALTER TABLE oskari_maplayer_externalid ADD FOREIGN KEY (maplayerid) REFERENCES oskari_maplayer (id) ON DELETE CASCADE;
+ALTER TABLE oskari_maplayer_externalid ALTER externalid SET NOT NULL;
```

Comment:
External id is not unique at the moment, but I think it should be.

Reply:
Good idea. It should definitely be unique (as is assumed by the migrations in this PR for example).